### DISCUSSION TITLE
Can you sync applications created through the ui/cli back to a git repo?

### DISCUSSION THREAD

**Discussion by justyns** [argoproj/argo-cd#18771] [Q&A]
I've been reading through the docs and couldn't figure out if this is possible yet.   Can you create applications through the argocd cli and web ui, and then have argocd automatically sync those application definitions back to a git repo?

E.g.  I want a git repo to be the source of truth for the apps being managed by argocd, but sometimes I want to add a new app (or change an existing one) through the ui and have argocd sync that change back to the git repo.  Basically treating the git repo like a backup, but also letting other automation tools make changes to the git repo too.

**Comment by ChristianCiach** [comment]
I am pretty sure the answer to your question is "no".

But maybe it won't be too hard to build something like this by yourself? I am thinking about using Argo Events to watch [Kubernetes resources](https://argoproj.github.io/argo-events/eventsources/setup/resource/) and then use a Sensor to execute some kind of Trigger, like maybe a [HTTP Trigger](https://argoproj.github.io/argo-events/sensors/triggers/http-trigger/) to *directly* push the watched resource to the [Gitlab repository API](https://docs.gitlab.com/ee/api/repository_files.html#update-existing-file-in-repository) by utilizing [parameterization](https://argoproj.github.io/argo-events/tutorials/02-parameterization/) to construct the upload URL from the event.

I've never done something like that, but by reading the docs this seems possible.

**Comment by djjudas21** [comment]
There's no easy way, but you can get the YAML definition of an Argo application with this command, and then shove that back in a git repo. It will probably need some cleanup.

```console
kubectl get application appname -o yaml
```
